nsm/error.go

```diff
@@ -11,119 +9,108 @@ var (
 }
 
 // ErrCreatingHelmIndex is the error for creating helm index
 func ErrCreatingHelmIndex(err error) error {
-	return errors.NewDefault(ErrInstallNSMCode, fmt.Sprintf("Error with nsm operation: %s", err.Error()))
+	return errors.New(ErrInstallNSMCode, errors.Alert, []string{"Error with nsm operation: ", err.Error()}, []string{}, []string{}, []string{})
 }
 
 // ErrEntryWithAppVersionNotExists is the error when an entry with the given app version is not found
 func ErrEntryWithAppVersionNotExists(entry, appVersion string) error {
-	return errors.NewDefault(
-		ErrEntryWithAppVersionNotExistsCode,
-		fmt.Sprintf("entry %s with app version %s does not exists", entry, appVersion),
-	)
+	return errors.New(ErrEntryWithAppVersionNotExistsCode, errors.Alert, []string{"entry: ", entry, "with app version : ", appVersion, "does not exists"}, []string{}, []string{}, []string{})
 }
 
 // ErrHelmRepositoryNotFound is the error when no valid remote helm repository is found
 func ErrHelmRepositoryNotFound(repo string, err error) error {
-	return errors.NewDefault(
-		ErrHelmRepositoryNotFoundCode,
-		fmt.Sprintf("either the repo %s does not exists or is corrupt: %v", repo, err),
-	)
+	return errors.New(ErrHelmRepositoryNotFoundCode, errors.Alert, []string{"either the repo : ", repo, "does not exists or is corrupt: ", err.Error()}, []string{}, []string{}, []string{})
 }
 
 // ErrDecodeYaml is the error when the yaml unmarshal fails
 func ErrDecodeYaml(err error) error {
-	return errors.NewDefault(
-		ErrDecodeYamlCode,
-		fmt.Sprintf("error decoding yaml: %v", err),
-	)
+	return errors.New(ErrDecodeYamlCode, errors.Alert, []string{"error decoding yaml: ", err.Error()}, []string{}, []string{}, []string{})
 }
 
 // ErrMeshConfig is the error for mesh config
 func ErrMeshConfig(err error) error {
-	return errors.NewDefault(ErrMeshConfigCode, fmt.Sprintf("Error configuration mesh: %s", err.Error()))
+	return errors.New(ErrMeshConfigCode, errors.Alert, []string{"Error configuration mesh: ", err.Error()}, []string{}, []string{}, []string{})
 }
 
 // ErrApplyHelmChart is the error for applying helm chart
 func ErrApplyHelmChart(err error) error {
-	return errors.NewDefault(ErrApplyHelmChartCode, fmt.Sprintf("error applying helm chart: %s", err.Error()))
+	return errors.New(ErrApplyHelmChartCode, errors.Alert, []string{"error applying helm chart: ", err.Error()}, []string{}, []string{}, []string{})
 }
 
 // ErrConvertingAppVersionToChartVersion is the error for converting app version to chart version
 func ErrConvertingAppVersionToChartVersion(err error) error {
-	return errors.NewDefault(
-		ErrConvertingAppVersionToChartVersionCode,
-		fmt.Sprintf("error converting app version to chart version: %s", err.Error()),
-	)
+	return errors.New(ErrConvertingAppVersionToChartVersionCode, errors.Alert, []string{"error converting app version to chart version: ", err.Error()}, []string{}, []string{}, []string{})
```

Comment:
can't remember the outcome of our capitalization discussion? should the first letter be a capital letter?

Reply:
Yes @mgfeller I forget this everytime my bad.